app/controllers/management/site_pages_controller.rb

```diff
@@ -96,6 +96,11 @@ def toggle_enable
   def set_site
     @site = Site.find_by({slug: params[:site_slug]})
 
+    unless @site.present?
```

Comment:
too much work to get a 404, Rails will handle this automatically if correct finder is used (https://github.com/Vizzuality/forest-atlas-landscape-cms/pull/230)